compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/cfa/CfaUtils.kt

```diff
@@ -60,21 +59,59 @@ class PropertyInitializationInfo(
         ::EMPTY
 }
 
-class LocalPropertyCollector private constructor() : ControlFlowGraphVisitorVoid() {
+class LocalPropertyAndCapturedWriteCollector private constructor() : ControlFlowGraphVisitorVoid() {
     companion object {
-        fun collect(graph: ControlFlowGraph): MutableSet<FirPropertySymbol> {
-            val collector = LocalPropertyCollector()
+        fun collect(graph: ControlFlowGraph): Pair<Set<FirPropertySymbol>, Set<FirVariableAssignment>> {
+            val collector = LocalPropertyAndCapturedWriteCollector()
             graph.traverse(TraverseDirection.Forward, collector)
-            return collector.symbols
+            return collector.symbols.keys to collector.capturedWrites
         }
     }
 
-    private val symbols: MutableSet<FirPropertySymbol> = mutableSetOf()
+    // Mapping from a property symbol to its declaration context
+    // `null` if the (local) property is declared in the currently visited function.
+    private val symbols: MutableMap<FirPropertySymbol, FirFunction<*>?> = mutableMapOf()
```

Comment:
The value of this map is never used. Use a set instead?

Reply:
It's actually used at line 106. So the value of this map is used to determine whether the property of interest is defined in the outer-most function or inner lambda/local function. I initially double-checked the last lambda/local function in the stack matches with the value of this map, and found that knowing that it's in the current outer-most function or not is good enough. So, definitely set isn't a solution, but I have to admit map isn't the best solution either (or is it?).

Reply:
I see, how about change line 81 to only assign if not null?

Reply:
Note that, previously, all local properties, including those declared in a lambda/local function, are returned to the caller (analyzer component) so that they know what properties to collect init info. And, after changing this to a map, `keys` is used as that source.

If we assign a mapping if the value is not null, that means we rather skip local properties that are declared in the outer-most function in this context. If we're using some other conditions, like if the value is null, then we may skip those declared in a lambda/local function.

Since it's good enough to know whether this local variable isn't defined in a lambda/local function, maybe the value could be simply enum or just boolean. Will it be readable/helpful?